reconcile/test/test_utils_terraform_client.py

```diff
@@ -251,3 +251,258 @@ def test__resource_diff_changed_fields(tf):
     )
 
     assert changed == set()
+
+
+def test_get_pending_rds_modification(aws_api, tf):
```

Comment:
Thank you for adding these tests! A couple of minor suggestions:

* You can take a look at the pytest documentation for [parametrizing test functions](https://docs.pytest.org/en/6.2.x/parametrize.html). This will run a separate test per input, which has some benefits including that a single failed assert won't mean the rest of your cases don't run, and in some cases can arguably make the tests easier to read.
* There's generally a convention of having separate functions per test case and making the names fairly descriptive. For instance, at a minimum you could break out the exception case and call it `test_get_pending_rds_modifications_raise_on_error`. With that, you'd probably reduce this all to 2 tests, one having many parameters, but very little actual code.

Reply:
Done. Thank you!